lib/Script/Expression.cpp

```diff
@@ -726,7 +726,11 @@ eld::Expected<uint64_t> AlignOf::evalImpl() {
   // sections are known only after all the assignments are complete
   if (ThisSection == nullptr)
     ThisSection = ThisModule.getScript().sectionMap().find(Name);
-  assert(ThisSection != nullptr);
+  if (!ThisSection) {
+    ThisModule.getConfig().raise(Diag::sect_invalid) << Name;
```

Comment:
Return the error as `DiagnosticEntry` instead, for this function return value of `0` does not represent failure.

Reply:
Have made these changes